### ISSUE TITLE
Fails to load ELF files from Android

### ISSUE THREAD

**Issue by bruno-garcia** [xoofx/LibObjectFile#5]
Is it expected that the library can load ELF files from an Android emulator?

The files in question came from an Android emulator (x86) but I'm p[rocessing them via a .NET Core 3.1 console app](https://github.com/bruno-garcia/elf-parsing-repro/tree/master/Console). ELFSharp can successfully load the same files.

Example output:

```
Processing file: Files/libchrome.so
ELF diag Error LB0114: Unable to read program header table as the size of program header entry (e_phentsize) == 0 in the Elf Header
Error LB0118: Unable to read section header table as the size of section header entry (e_ehsize) == 0 in the Elf Header
```

Please note these are different errors than when the same code runs on Android. `LibObjectFile` in Android seem not to work for other reasons likely unrelated to the library itself. 

[One of such files is here](https://github.com/bruno-garcia/elf-parsing-repro/blob/master/Console/Files/libdl.so). It's part of a repro I've made [which includes a readme](https://github.com/bruno-garcia/elf-parsing-repro). It also includes more details relevant to `Xamarin.Android` that are not related to this issue.

**Comment by xoofx** [comment]
It's just that the test might be to restrictive for that field. PR welcome.

**Comment by bruno-garcia** [comment]
While investigating this, lead me to open #8 .
`ELFSharp` chooses whether to use the DirectDecoder or the endianness flipping one based on the Class while LibObjectFIle does that based on the 5th bit (encoding).

I successfully parsed ELF files from Linux x86, ARM and Android using that library so I'm afraid the problem is here.

Also, the output of loading the same library when running on an Android emulator and on my desktop differs. If I force (with the debugger) to use Class to define endienness, the output matches in both.

**Comment by bruno-garcia** [comment]
After the bug fix released on `0.3.2` the behavior on Android and .NET Core 3 match.

In both cases it fails with:
<img width="1256" alt="image" src="https://user-images.githubusercontent.com/1633368/71323477-f11b0280-24d3-11ea-8ce1-8484390645f5.png">

Closing this issue now.

**Comment by xoofx** [comment]
Could you open an issue with attached the ELF file generating such errors?, I'm curious how this can happen.
